titan/features/incar.py

```diff
@@ -0,0 +1,202 @@
+
+        args:
+            pop: the population this agent is a part of
+            time: the current time step
+        """
+        incar_params = self.agent.location.params.demographics[self.agent.race][
+            self.agent.sex_type
+        ].incar
+        jail_duration = incar_params.duration.init
+
+        prob_incar = incar_params.init
+        if pop.pop_random.random() < prob_incar:
+            self.active = True
+            bin = 1
+            current_p_value = jail_duration[bin].prob
+            p = pop.pop_random.random()
+
+            while p > current_p_value:
+                bin += 1
+                current_p_value += jail_duration[bin].prob
+
+            self.time = pop.pop_random.randrange(
+                jail_duration[bin].min, jail_duration[bin].max
+            )
+
+    def update_agent(self, model: "model.HIVModel"):
+        """
+        Update the agent for this feature for a time step.  Called once per time step in `HIVModel.update_all_agents`. Agent level updates are done after population level updates.   Called on only features that are enabled per the params.
+
+        Incarcerate an agent or update their incarceration variables
+
+        args:
+            model: the instance of HIVModel currently being run
+        """
+        hiv_bool = self.agent.hiv
+
+        if hiv_bool:
+            hiv_multiplier = self.agent.location.params.incar.hiv.multiplier
+        else:
+            hiv_multiplier = 1
+
+        # agent is incarcerated
+        if self.active:
+            self.time -= 1
+
+            # FREE AGENT
+            if self.time == 0:
+                self.add_agent(self.agent)
+                self.active = False
+
+                # become high risk on release
+                if (
+                    not self.agent.high_risk.active and model.params.features.high_risk  # type: ignore[attr-defined]
+                ):  # If behavioral treatment on and agent HIV, ignore HR period.
+                    self.agent.high_risk.become_high_risk()  # type: ignore[attr-defined]
+                    for bond in self.agent.location.params.high_risk.partnership_types:
+                        self.agent.mean_num_partners[
+                            bond
+                        ] += self.agent.location.params.high_risk.partner_scale
+                        self.agent.target_partners[bond] = utils.poisson(
+                            self.agent.mean_num_partners[bond], model.np_random
+                        )
+                    model.pop.update_partnerability(self.agent)
+
+                # does agent stay on haart
+                if hiv_bool:
+                    if self.agent.haart.active:  # type: ignore[attr-defined]
+                        if (
+                            model.run_random.random()
+                            <= self.agent.location.params.incar.haart.discontinue
+                        ):  # 12% remain surpressed
```

Comment:
```suggestion
                        ):
```

That number is params-defined now.